packages/react/src/internal/useResizeObserver.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useRef, useState, useLayoutEffect, useEffect } from 'react';
+
+export const useResizeObserver = ({ ref, onResize, deps = [] }) => {
```

Comment:
```suggestion
export const useResizeObserver = ({ ref, onResize }) => {
```

If we don't have a need for this, I'd leave it out. Makes it really easy for someone to accidentally have these useEffects re-run on every render. If there's a solid usecase for it though I think it'd make sense to write a test for it. Sort of "documenting" that use case via the test, if that makes sense. Non blocking, just my 2c

Reply:
Agreed, I removed it since there currently isn't a need.